src/main/java/pl/coderslab/warsztat3katw03/dao/ExerciseDAO.java

```diff
@@ -0,0 +1,160 @@
+package pl.coderslab.warsztat3katw03.dao;
+
+import pl.coderslab.warsztat3katw03.db.DbUtil;
+import pl.coderslab.warsztat3katw03.model.Exercise;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExerciseDAO {
+    private static final String CREATE_EXERCISE_QUERY = "INSERT INTO exercise(title, description) VALUES (?, ?)";
+    private static final String READ_EXERCISE_QUERY = "SELECT title, description FROM exercise where id = ?";
+    private static final String UPDATE_EXERCISE_QUERY = "UPDATE exercise SET title = ?, description = ? where id = ?";
+    private static final String DELETE_EXERCISE_QUERY = "DELETE FROM exercise WHERE id = ?";
+    private static final String FIND_ALL_EXERCISES_QUERY = "SELECT id, title, description FROM exercise";
+    private static final String FIND_ALL_EXERCISES_WITHOUT_SOLUTION_QUERY = "SELECT e.id, title, e.description FROM exercise e LEFT JOIN solution s ON e.id = s.exercise_id WHERE s.exercise_id != (SELECT s.exercise_id FROM solution WHERE s.user_id = ? GROUP BY s.exercise_id) AND s.user_id != ? OR S.exercise_id IS NULL";
+    private static final String FIND_EXERCISE_WITH_SOLUTION_OF_USER_QUERY = "SELECT e.id, title, e.description FROM exercise e LEFT JOIN solution s ON e.id = s.exercise_id WHERE s.user_id = ? AND s.exercise_id = ?";
+    private static final String FIND_LAST_EXERCISEID_QUERY = "SELECT id, title, description FROM exercise ORDER BY id DESC LIMIT 1";
+
+    public static Exercise create(Exercise exercise) {
+        try  {
+            Connection conn = DbUtil.getConnection();
+            PreparedStatement statement = conn.prepareStatement(CREATE_EXERCISE_QUERY, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, exercise.getTitle());
```

Comment:
sugeruje użycie indeksu inkrementowanego jak to sugerowałem w poprzednich warsztatach 
```
++idx, exercise.getTItle());
```